### PR TITLE
Map 119 update msg builder with two optional fields

### PR DESCRIPTION
**Pull request by ChrisC027**
# PR Details
## Description

Once after adding Node Speed Limit and Region ID optional fields in the UI, implement these new fields in the message builder to encode the map message. Make sure that the map message is encoded with the two optional fields.

## Related Jira Key

MAP-119

## Motivation and Context

Currently, the two fields are present in the UI, but are not being encoded in the message builder. 

## How Has This Been Tested?

Built and tested locally using various speed limit types, velocities, and region ids.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: fedgov-cv-lib-asn1c/bin/third_party_lib/libasn1c.so
Comment (paulbourelly999) [inline fedgov-cv-lib-asn1c/bin/third_party_lib/libasn1c.so:1]: Are these files meant to be committed

Comment (DokurOmkar) [reply fedgov-cv-lib-asn1c/bin/third_party_lib/libasn1c.so:1]: Yeah, I think you can unstage the lib-asn1c package as none of these files need to be touched.

File: fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java [hunk 1]
```diff
@@ -24,18 +24,28 @@ public class LaneDataAttribute {
...
+    public static final int LANECROWNPOINTRIGHT = 3;
+    public static final int LANECROWNPOINTLEFT = 4;
+    public static final int LANEANGLE = 5;
+    public static final int SPEEDLIMITS = 6;
 
     public LaneDataAttribute() {
     }
 
-    public LaneDataAttribute(int choice, int pathEndPointAngle, int laneCrownPointCenter, int laneCrownPointRight, int laneCrownPointLeft, int laneAngle, SpeedLimitList speedLimits) {
+    public LaneDataAttribute(int choice, int pathEndPointAngle, int laneCrownPointCenter, int laneCrownPointRight, int laneCrownPointLeft, int laneAngle, SpeedLimitList speedLimits, boolean speedLimitsExist) {
```
Comment (DokurOmkar) [inline fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java:40]: I think we can remove this Boolean as we are not going to use it

**Review by DokurOmkar — CHANGES_REQUESTED** [review]

File: fedgov-cv-lib-asn1c/bin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
[the remark this answers is not in the document]
Comment (ChrisC027) [reply fedgov-cv-lib-asn1c/bin/pom.xml]: Resolved

File: fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java (hunk 1 above)
Comment (ChrisC027) [reply fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java]: Resolved

File: fedgov-cv-lib-asn1c/bin/third_party_lib/libasn1c.so
Comment (ChrisC027) [reply fedgov-cv-lib-asn1c/bin/third_party_lib/libasn1c.so:1]: @paulbourelly999 I have removed them.

**Review by DokurOmkar — APPROVED** [review]

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Will approve once Sonar Code Smells are addressed.

File: fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/input/IntersectionInputData.java [hunk 3]
```diff
@@ -275,7 +275,7 @@ public static class ReferencePointChild {
 		public RegulatorySpeedLimit[] speedLimitType;	
 	}
 	
-	// @JsonTypeName("speedLimitType")
+	// @JsonTypeName("speedLimitType")
```
Comment (paulbourelly999) [inline fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/input/IntersectionInputData.java:278]: NIT: In the future please remove any commented out code that is no longer being used. Leaving commented out sections in our code makes it harder to read especially when there is not comment for context.

File: fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java [hunk 4]
```diff
@@ -25,6 +25,14 @@ public class LaneDataAttribute {
     private int laneAngle;
     private SpeedLimitList speedLimits;
 
+    //enum choices
+    public static final int PATH_END_POINT_ANGLE = 1;
```
Comment (paulbourelly999) [inline fedgov-cv-mapencoder/src/main/java/gov/usdot/cv/mapencoder/LaneDataAttribute.java:29]: Is there a reason we did not make this a regular java enumeration ? Something like 
```java
public enum LaneDataAttributeEnum {
    PATH_END_POINT_ANGLE(1),
    LANE_CROWN_POINT_CENTER(2),
    LANE_CROWN_POINT_RIGHT(3),
    LANE_CROWN_POINT_LEFT(4),
    LANE_ANGLE(5),
    SPEED_LIMITS(6);
    private int value;
    LaneDataAttributeEnum( int value) {
        this.value=value;
    }
    public int getValue() { return this.value;}
   
}
```

File: fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/IntersectionSituationDataBuilder.java [hunk 5]
```diff
@@ -581,6 +591,31 @@ private NodeListXY buildNodeList(IntersectionInputData isdInputData, DrivingLane
 					}
 				}
 
+				if (laneNode.speedLimitType != null && laneNode.speedLimitType.length > 0) {
```
Comment (paulbourelly999) [inline fedgov-cv-message-builder/src/main/java/gov/usdot/cv/msg/builder/IntersectionSituationDataBuilder.java:594]: For code segments like this inline documentation can help new developers or developers returning to this logic understand the logic and intention better

**Review by paulbourelly999 — APPROVED** [review]
Left some NIT comments. In the future please consider these.

**Pull request merged**